src/parser/handler.py

```diff
@@ -21,7 +21,18 @@ def lambda_handler(event, context):
         if diff['configurationItemDiff']:
             for k,v in diff['configurationItemDiff']['changedProperties'].items():
                 if 'BlockDeviceMappings' in k:
-                    _logger.info(f'The EBS update was marked as spam. Skip execution.')
+                    _logger.info(f'The EBS attaching was marked as spam. Skip execution.')
+                elif 'Relationships' in k:
+                    try:
+                        if v['previousValue']['resourceType'] == "AWS::EC2::Volume":
```

Comment:
it seems you can omit one if statement and use 
`if v['previousValue']['resourceType'] == "AWS::EC2::Volume" or v['updatedValue']['resourceType'] == "AWS::EC2::Volume"`
correct me if i'm wrong

Reply:
also, what are the conditions you get the TypeError?

Reply:
I did it before, but we actually can't. The problem comes with the input event. Look at this json

```json
"Relationships.0": {
        "previousValue": {
          "resourceId": "vol-0cf23d22b71819fa3",
          "resourceName": null,
          "resourceType": "AWS::EC2::Volume",
          "name": "Is attached to Volume"
        },
        "updatedValue": null,
        "changeType": "DELETE"
      },
```

You can see that one of the `updatedValue` or `previousValue` is null. So, I will get TypeError in this case (that's why I've added `try` `except` block here). But ONLY one `updateValue` or `previousValue` can be null at the same time. So, if I put it under the single `try` `except` block the `TypeError` may happen for the first arg `v['previousValue']['resourceType']` in `if` block, so condition will never check the second arg `v['updatedValue']['resourceType']`

Reply:
> also, what are the conditions to get the TypeError?

if `v['previousValue']` is null I will get `TypeError` when referring to `v['previousValue']['resourceType']` (when go deeper)

Reply:
oh, in this case you should use v.get(key, default_value) to omit errors

Reply:
oh, really

Reply:
okay. @eyehol3 `default_value` won't fix the problem because I don't need to set the default value. If the value is null, it's okay for me to move forward to the next check step

Reply:
We shouldn't guess whether `previousValue` or `updatedValue` exist. It depends on the changeType and we know it based on it. There are just 3 cases: create, update, delete. I suggest, we should have a 
```
if changeType == 'DELETE':
    ...
elif changeType == 'UPDATE':
   ...
elif changeType == 'CREATE':
```

It provides better understanding of the code in my opinion, and avoids using excessive try/excepts

Reply:
Also, this check seems to be pretty universal. So if we need to "whitelist" something else in addition to EBS volume, we can simply update the if statement. so, why not move it to a different function and do the inclusive check of `v['previousValue']['resourceType'] in list[EBS,whatever]` instead?

Reply:
@jradikk prepared an update according your comments